modules/andes-core/broker/src/main/java/org/wso2/andes/subscription/SubscriptionBitMapHandler.java

```diff
@@ -0,0 +1,671 @@
+                 * 2 constituentsAdd a new table for the constituent part 3
+                 */
+                if (bitMapLocal.size() <= constituentPart) {
+                    Map<String, BitSet> newBitMapForIthConstituent;
+                    newBitMapForIthConstituent = Collections.synchronizedMap(new HashMap<String, BitSet>());
+
+                    BitSet bitSet;
+                    bitSet = new BitSet(localSubscriptionCount);
+                    bitSet.set(0, localSubscriptionCount);
+
+                    for (int deleted : deletedLocals) bitSet.clear(deleted);
+                    bitSet.clear(columnIndexOfTheSubscriptionInBitMap);
+                    newBitMapForIthConstituent.put(SPECIAL_CHARACTER_FOR_NULL, bitSet);
+
+                    BitSet bitSetForPreviousOther;
+                    bitSetForPreviousOther = new BitSet();
+
+                    if (constituentPart != 0) {
+                        bitSetForPreviousOther = (BitSet) bitMapLocal
+                                .get(constituentPart - 1).get(CONSTITUENT_TOPIC_CONSTANT).clone();
+                        BitSet bitArrayForPreviousNull = (BitSet) bitMapLocal
+                                .get(constituentPart - 1).get(SPECIAL_CHARACTER_FOR_NULL).clone();
+
+                        bitSetForPreviousOther.and(bitArrayForPreviousNull);
+                    }
+
+                    BitSet bitSetForOther = bitSetForPreviousOther;
+                    newBitMapForIthConstituent.put(CONSTITUENT_TOPIC_CONSTANT, bitSetForOther);
+                    bitMapLocal.add(newBitMapForIthConstituent);
+                }
+
+                /** Set the bit for the subscription constituent part
+                 *
+                 */
+                Map<String, BitSet> bitMapForithConstituent = bitMapLocal.get(constituentPart);
+
+                if ("#".equals(destinations[constituentPart]) || "*".equals(destinations[constituentPart])) {
+                    for (Map.Entry<String, BitSet> mapEntryForithBitMap : bitMapLocal.get(constituentPart).entrySet()) {
+                        if (!SPECIAL_CHARACTER_FOR_NULL.equals(mapEntryForithBitMap.getKey())) {
+                            mapEntryForithBitMap.getValue().set(columnIndexOfTheSubscriptionInBitMap);
+                        } else {
+                            if ("#".equals(destinations[constituentPart]))
+                                mapEntryForithBitMap.getValue().set(columnIndexOfTheSubscriptionInBitMap);
+                        }
+                    }
+                } else {
+                    BitSet bitSetForspecificConstituentPartOfSubscription;
+                    bitSetForspecificConstituentPartOfSubscription = bitMapForithConstituent.get(destinations[constituentPart]);
+                    if (bitSetForspecificConstituentPartOfSubscription == null)
+                        bitSetForspecificConstituentPartOfSubscription = (BitSet) bitMapLocal.get(constituentPart).get("other")
```

Comment:
a magical static string "other". What about defining a constant? and comment on specialty of "other"?